run_assistant.py

```diff
@@ -104,4 +105,4 @@ def main():
             p.terminate()
```

Comment:
do we still need this here?

Reply:
If we leave the pyaudio object open too long, it can cause the program to crash in my experience. It's possible closing the stream has fixed this, I haven't tested it really, but that's my logic for having that there.

Reply:
I was unclear, i was referring to the fact that p is terminated twice, once on line 97 and then 105

Reply:
Ahh yes, you're right. You only need the first call at line 97.

Reply:
Oh. You're right, we only need the one. I'll fix that!